### PR TITLE
Modify the compressed FITS image loader destruction

### PR DESCRIPTION
**Pull request by markccchiang**
I think it can solve #999 and #1014. When doing statistics calculations, several copies of an image loader are created and run in parallel. These copies are deleted when calculations are done. In CartaFitsImage, the close file action can only be executed when all of the CartaFitsImage objects are destroyed.

### CODE REVIEW

**Comment by kswang1029** [comment]
@markccchiang confirmed this PR does not result in any regressions that the current e2e tests are capable to capture. Manual tests are on the way before approving the PR.

**Review by kswang1029 — APPROVED** [review]
tested manually with compressed fits and confirmed that region statistics, line/polyline spatial profiler, region histogram, pv generator and region spectral profile work and do not trigger crash.

**Comment by pford** [comment]
Keeping track of copies sounds like a shared_ptr workaround.  But the real issue is several CartaFitsImages sharing a single fitsfile* pointer.  Maybe instead the copy should have a null _fptr and it will open the file when it needs to read something?

File: src/ImageData/CartaFitsImage.cc [hunk 1]
```diff
@@ -54,15 +55,18 @@ CartaFitsImage::CartaFitsImage(const CartaFitsImage& other)
       _datatype(other._datatype),
       _has_blanks(other._has_blanks),
       _pixel_mask(nullptr),
-      _tiled_shape(other._tiled_shape) {
+      _tiled_shape(other._tiled_shape),
+      _is_copy(true) {
```
Comment (pford) [inline src/ImageData/CartaFitsImage.cc:59]: Set _fptr(nullptr) instead?

**Review by pford — CHANGES_REQUESTED** [review]

**Comment by pford** [comment]
It occurred to me that opening the same gz file in another CartaFitsImage may decompress it into memory again.  Can you test this?  Your solution was probably better.

**Comment by markccchiang** [comment]
> It occurred to me that opening the same gz file in another CartaFitsImage may decompress it into memory again. Can you test this? Your solution was probably better.

A quick test with `psrecord`. The memory usage for statistics calculations with multiple `_fptr` is 

test3

On the other hand, the memory usage for statistics calculations with a single `_fptr` is 

test2

So maybe it is better to use the original method?

**Comment by markccchiang** [comment]
> It occurred to me that opening the same gz file in another CartaFitsImage may decompress it into memory again. Can you test this? Your solution was probably better.

A quick test with `psrecord`. The memory usage for statistics calculations with multiple `_fptr` is 

test3

On the other hand, the memory usage for statistics calculations with a single `_fptr` is 

test2

So maybe it is better to use the original method?

**Comment by pford** [comment]
Yes, please put it back to the tested version.  Sorry about that, I forgot about decompression!

**Review by pford — APPROVED** [review]

**Pull request merged**